src/engine/renderer/tr_video.cpp

```diff
@@ -0,0 +1,849 @@
+/*
+===========================================================================
+
+Daemon GPL Source Code
+Copyright (C) 1999-2010 id Software LLC, a ZeniMax Media company.
+
+This file is part of the Daemon GPL Source Code (Daemon Source Code).
+
+Daemon Source Code is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Daemon Source Code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Daemon Source Code.  If not, see <http://www.gnu.org/licenses/>.
+
+In addition, the Daemon Source Code is also subject to certain additional terms.
+You should have received a copy of these additional terms immediately following the
+terms and conditions of the GNU General Public License which accompanied the Daemon
+Source Code.  If not, please request a copy in writing from id Software at the address
+below.
+
+If you have questions concerning this license or the applicable additional terms, you
+may contact in writing id Software LLC, c/o ZeniMax Media Inc., Suite 120, Rockville,
+Maryland 20850 USA.
+
+===========================================================================
+*/
+
+// tr_video.cpp: ROQ video playback on in-game surfaces
+//
+// This is a greatly cut-down version of the ROQ support in Quake 3. Cinematic mode has
+// been removed, along with a lot of dead or broken code. If something with in-game videos
+// works with Q3 but not Daemon, one may consult the ioq3 implementation:
+// https://github.com/ioquake/ioq3/blob/main/code/client/cl_cin.c
+// Variable and function names have been kept close to the original, even when the names
+// don't make much sense, to assist in the comparison.
```

Comment:
If you'd like to get better function names, what you can do is to add the ioq3 function name in a comment in the previous line for reference?

Reply:
Yeah that would work. I'm not going to do further rewriting at this point in time though.